pkg/apis/config/sigstore_keys.go

```diff
@@ -0,0 +1,184 @@
+//
+// Copyright 2022 The Sigstore Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package config
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/sigstore/policy-controller/pkg/apis/policy/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
+	"knative.dev/pkg/apis"
+	"sigs.k8s.io/yaml"
+)
+
+const (
+	// SigstoreKeysConfigName is the name of ConfigMap created by the
+	// reconciler and consumed by the admission webhook for determining
+	// which Keys/Certificates are trusted for things like Fulcio/Rekor, etc.
+	SigstoreKeysConfigName = "config-sigstore-keys"
+)
+
+// Note that these are 1:1 mapped to public API SigstoreKeys. Reasoning
+// being that we may choose to serialize these differently, or use the protos
+// that are defined upstream, so want to keep the public/private distinction, so
+// that we can change things independend of breaking the API. Time will tell
+// if this is the right call, but we can always reunify them later if we so
+// want.
+
+// TransparencyLogInstance describes the immutable parameters from a
+// transparency log.
+// See https://www.rfc-editor.org/rfc/rfc9162.html#name-log-parameters
+// for more details.
+// The incluced parameters are the minimal set required to identify a log,
+// and verify an inclusion promise.
+type TransparencyLogInstance struct {
+	BaseURL       apis.URL `json:"baseURL"`
+	HashAlgorithm string   `json:"hashAlgorithm"`
+	// PEM encoded public key
+	PublicKey []byte `json:"publicKey"`
+	LogID     string `json:"logID"`
+}
+
+type DistinguishedName struct {
+	Organization string `json:"organization"`
+	CommonName   string `json:"commonName"`
+}
+
+type CertificateAuthority struct {
+	// The root certificate MUST be self-signed, and so the subject and
+	// issuer are the same.
+	Subject DistinguishedName `json:"subject"`
+	// The URI at which the CA can be accessed.
+	URI apis.URL `json:"uri"`
+	// The certificate chain for this CA.
+	// CertChain is in PEM format.
+	CertChain []byte `json:"certChain"`
+
+	// TODO(vaikas): How to best represent this
+	// The time the *entire* chain was valid. This is at max the
+	// longest interval when *all* certificates in the chain where valid,
+	// but it MAY be shorter.
+	//       dev.sigstore.common.v1.TimeRange valid_for = 4;
```

Comment:
Wouldn't this be a TTL ?

Reply:
I think the problem is that it will require a start and optionally an end :) So, I'd like to figure out how this should work hence the TODO :)